extensions/replication/tasks/MultipleBackendTask.js

```diff
@@ -825,7 +799,26 @@ class MultipleBackendTask extends ReplicateObject {
                     return this._deleteObjectTagging(sourceEntry, destEntry,
                         log, next);
                 }
-                return this._getAndPutData(sourceEntry, destEntry, log, next);
+                // Do a multipart upload when either the size is above
+                // a threshold or the source object is itself a MPU.
```

Comment:
What is the rationale for the 20MB threshold?

Reply:
It's a heuristic, if the object is "big enough" we want to MPU it to benefit from more parallelism (and later maybe better recovery on error by keeping the valid parts around), if the object is too small it's overkill to do a MPU (more requests and metadata overhead). So we need some boundary to decide between the two, 20MB makes a MPU of 4x5MB parts for example, which sounds like a reasonable size to start thinking about MPU.

Of course the size threshold can be debated, it just looks a reasonable tradeoff to me.